datafusion/core/src/physical_optimizer/pruning.rs

```diff
@@ -1315,24 +1351,23 @@ const MAX_LIST_VALUE_SIZE_REWRITE: usize = 20;
 /// Translate logical filter expression into pruning predicate
 /// expression that will evaluate to FALSE if it can be determined no
 /// rows between the min/max values could pass the predicates.
+/// 
+/// Any predicates that can not be translated will be passed to `unhandled_hook`.
 ///
 /// Returns the pruning predicate as an [`PhysicalExpr`]
 ///
-/// Notice: Does not handle [`phys_expr::InListExpr`] greater than 20, which will be rewritten to TRUE
-fn build_predicate_expression(
+/// Notice: Does not handle [`phys_expr::InListExpr`] greater than 20, which will fall back to calling `unhandled_hook`
+pub fn build_predicate_expression(
```

Comment:
I'm open to new names. We could also make a wrapper that e.g. does not require `required_columns` and takes `Option<UnhandledPredicateHook>` which defaults to `true`.